state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1352,6 +1354,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		tx_id: <T::TargetChain as Chain>::DepositDetails,
 	) -> DispatchResult {
 		let account_id = T::AccountRoleRegistry::ensure_broker(origin)?;
+		ensure!(
+			!TaintedTransactions::<T, I>::contains_key(&account_id, &tx_id),
```

Comment:
I don't think this is enough? We don't need to know if it has been reported, we need to know if it has been boosted.

For example, what about a case where a transaction has been boosted (without being reported)?